packages/cli-lib/lib/uploadFolder.js

```diff
@@ -28,43 +29,100 @@ const queue = new PQueue({
   concurrency: 10,
 });
 
-function getFilesByType(files) {
-  const moduleFiles = [];
-  const cssAndJsFiles = [];
-  const otherFiles = [];
-  const templateFiles = [];
-  const jsonFiles = [];
-
-  files.forEach(file => {
-    const parts = splitLocalPath(file);
-    const extension = getExt(file);
-
-    const moduleFolder = parts.find(part => part.endsWith('.module'));
-    if (moduleFolder) {
-      moduleFiles.push(file);
-    } else if (extension === 'js' || extension === 'css') {
-      cssAndJsFiles.push(file);
-    } else if (extension === 'html') {
-      templateFiles.push(file);
-    } else if (extension === 'json') {
-      jsonFiles.push(file);
+const FileTypes = {
+  module: 'moduleFiles',
+  cssAndJs: 'cssAndJsFiles',
+  template: 'templateFiles',
+  json: 'jsonFiles',
+  other: 'otherFiles',
+};
+
+// Checks if file is in a module folder and if it is, returns the module folder name
+function isModuleFolder(filePath) {
+  const parts = splitLocalPath(filePath);
+  return parts.find(part => part.endsWith('.module'));
+}
+
+function getFileType(filePath) {
+  const extension = getExt(filePath);
+  const moduleFolder = isModuleFolder(filePath);
+  if (moduleFolder) return FileTypes.module;
+
+  switch (extension) {
+    case 'js':
+    case 'css':
+      return FileTypes.cssAndJs;
+    case 'html':
+      return FileTypes.template;
+    case 'json':
+      return FileTypes.json;
+    default:
+      return FileTypes.other;
+  }
+}
+
+function getFilesByType(filePaths, src, rootWriteDir, processFields) {
+  const srcDirRegex = new RegExp(`^${escapeRegExp(src)}`);
+  const fieldsJsObjects = [];
+  const filePathsByType = {
+    otherFiles: [],
+    moduleFiles: [],
+    cssAndJsFiles: [],
+    templateFiles: [],
+    jsonFiles: [],
+  };
+
+  filePaths.forEach(filePath => {
+    const fileType = getFileType(filePath);
+    const fileName = path.basename(filePath);
+    const relativePath = filePath.replace(srcDirRegex, '');
+
+    if (!processFields) {
+      filePathsByType[fileType].push(filePath);
+      return;
+    }
+    if (fileName === 'fields.output.json') return;
```

Comment:
I wonder if we could add this to the [CLI ignore rules](https://github.com/HubSpot/hubspot-cli/blob/75800573c73b7c08e16bd9bc132b9d3dd1e6b3e0/packages/cli-lib/ignoreRules.js) to always skip these files?